### PR TITLE
Ensure any axios requests also run the blacklist code for any redirects

### PR DESCRIPTION
**Pull request by jaywink**
Wrap axios.get in our own function with logic to check each redirect again the domain blacklist code.

To keep code more simple, do this consistently for all axios requests, even if calling the configured homeserver.

Moves the isDomainBlacklisted to utils.js.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -86,9 +90,110 @@ function requestLogger(req) {
...
+                return status >= 200 && status < 400;
+            },
+        },
+    );
+    if (response.status >= 300) {
+        if (redirects >= 4) {
+            // This was the fourth time following a redirect, abort
+            throw new Error('Maximum amount of redirects reached.');
+        }
+        return axiosGet(response.headers.location, redirects + 1);
```
Comment (dkasak) [inline src/utils.js]: Shouldn't the extra `headers` be repeated here?

File: tests/utils.tests.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            try {
+                await utils.axiosGet('http://127.0.0.1');
+                raised = false;
+            } catch (error) {
+                raised = true;
+            }
+            expect(raised).to.be.true;
+            expect(axiosStub.called).to.be.false;
+        });
+
```
Comment (dkasak) [inline tests/utils.tests.js:30]: I think a test that the redirection blacklisting is working as expected would be nice. Something like:

```suggestion

it('ensures redirection domain is not blacklisted', async() => {
    axiosStub = sinon.stub(axios, 'get');
                                                                 
    axiosStub.onFirstCall().returns({
        headers: {
            location: 'http://127.0.0.1',
        },
        status: 301,
    });
    axiosStub.onSecondCall().returns({status: 200});
                                                                 
    let raised;
    try {
        await utils.axiosGet('https://matrix.org');
        raised = false;
    } catch (error) {
        raised = true;
    }
                                                                 
    expect(raised).to.be.true;
    expect(axiosStub.calledOnce).to.be.true;
    expect(axiosStub.calledTwice).to.be.false;
});

```

**Review by dkasak — APPROVED** [review]
This generally looks good, but see my comment and additional test.

File: src/utils.js (hunk 1 above)
Comment (jaywink) [reply src/utils.js]: Yes! Good catch! :pray: Have patched.

**Pull request merged**